src/client/eppo-client.ts

```diff
@@ -1052,7 +1073,7 @@ export default class EppoClient {
   }
 
   private getFlag(flagKey: string): Flag | null {
-    return this.isObfuscated
+    return this.isObfuscated()
```

Comment:
every time a flag is evaluated will cause the warning above to be logged, which may or may not be desirable. I think i'd prefer it to be called once and its value cached, unless there's a reason not to?

Reply:
agreed that warning every time is too noisy. Recomputing every time gives us the ability to warn if the invalid state is entered sometime after the first call, but probably not worth the extra computation when the vast majority of use cases will not encounter this warning.
I've recoded to
1. cache whether the config is obfuscated; clearing that cache on a best effort basis (where the config is actually updated is not observable from the EppoClient, so we invalidate when a new flag config store is set, and when the config is fetched)
2. cache whether the client issued a warning about obfuscation mismatch. This cache is cleared when the configuration enters a good state (expected and actual obfuscation match), so that if a mismatched state is (re-)entered, a warning will still be issued.